crates/provider/src/provider/multicall/mod.rs

```diff
@@ -202,6 +202,23 @@ where
         self
     }
 
+    /// Add a dynamic call to the builder while specifying whether it is allowed to fail
+    ///
+    /// This is useful in combination with [`aggregate3`][MulticallBuilder::aggregate3].
+    pub fn add_dynamic3(
```

Comment:
this is the same as 

  pub fn add_dynamic(mut self, item: impl MulticallItem<Decoder = D>) -> Self {

but with the additional allow_failure?